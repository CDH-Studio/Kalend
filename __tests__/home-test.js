import React from "react";
import renderer from "react-test-renderer";
import {ImageBackground, Image, StatusBar} from "react-native";
import LinearGradient from 'react-native-linear-gradient';

// Removed this test in order to avoid the GoogleSignIn errors
// test("Home renders correctly", () => {
//   const tree = renderer.create(<Home />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

test("Linear Gradient renders correctly", () => {
    const linearGradient = renderer.create(<LinearGradient style={{flex: 1, width: '100%', height: '110%'}} colors={['#1473E6', '#0E55AA']}></LinearGradient>).toJSON();
    expect(linearGradient).toMatchSnapshot();
});

test("StatusBar renders correctly", () => {
    const statusBar = renderer.create(<StatusBar translucent={true} backgroundColor={'#00000050'}/>).toJSON();
    expect(statusBar).toMatchSnapshot();
});

test('Logo renders correctly', done => {
    Image.getSize('../assets/img/kalendFullLogo.png', (width, height) => {
      const logo = renderer.create(<Image style={{height, width}} />).toJSON();
      expect(logo).toMatchSnapshot();
      done();
    });
});

test('UserIcon renders correctly', done => {
    Image.getSize('../assets/img/loginScreen/userIcon.png', (width, height) => {
      const logo = renderer.create(<Image style={{height, width}} />).toJSON();
      expect(logo).toMatchSnapshot();
      done();
    });
});