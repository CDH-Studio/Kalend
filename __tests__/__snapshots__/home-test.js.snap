// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linear Gradient renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4279530470,
      4279129514,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "flex": 1,
      "height": "110%",
      "width": "100%",
    }
  }
/>
`;

exports[`Logo renders correctly 1`] = `
<Image
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
/>
`;

exports[`StatusBar renders correctly 1`] = `null`;

exports[`UserIcon renders correctly 1`] = `
<Image
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
/>
`;
